src/main/java/org/mule/service/http/impl/service/server/grizzly/ResponseStreamingCompletionHandler.java

```diff
@@ -45,15 +56,39 @@ public ResponseStreamingCompletionHandler(final FilterChainContext ctx, final Ht
     httpResponsePacket = buildHttpResponsePacket(request, httpResponse);
     inputStream = httpResponse.getEntity().getContent();
     memoryManager = ctx.getConnection().getTransport().getMemoryManager();
+    bufferSize = calculateBufferSize(ctx, httpResponse);
+    LOGGER.debug("Response streaming calculated buffer size = {} bytes.");
     this.responseStatusCallback = responseStatusCallback;
   }
 
+  private int calculateBufferSize(FilterChainContext ctx, HttpResponse httpResponse) {
+    String contentLengthHeader = httpResponse.getHeaderValue(CONTENT_LENGTH);
+    int contentLength;
+    if (!isEmpty(contentLengthHeader)) {
+      contentLength = Integer.valueOf(contentLengthHeader);
+    } else {
+      contentLength = -1;
+    }
+    if (contentLength > 0) {
```

Comment:
can the `content-Length` header have `0` as value? If not, you don't need the second `if` block